src/mds/nameserver2/chunk_allocator.h

```diff
@@ -42,6 +43,10 @@ class ChunkSegmentAllocator {
     virtual bool AllocateChunkSegment(FileType type,
         SegmentSizeType segmentSize, ChunkSizeType chunkSize,
         offset_t offset, PageFileSegment *segment) = 0;
+    virtual bool GetRemainingSpaceInLogicalPool(
+        std::vector<PoolIdType>* logicalPools,
+        std::map<PoolIdType, double>* enoughSpacePools,
+        bool chooseone) = 0;
```

Comment:
std::vector<PoolIdType>* logicalPools  is the input parameter，It is recommended to use const std::vector<PoolIdType> &logicalPools.